graph_api_client.go

```diff
@@ -0,0 +1,129 @@
+package azuresecrets
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/authorization/mgmt/2018-01-01-preview/authorization"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+const (
+	// defaultGraphMicrosoftComURI is the default URI used for the service MS Graph API
+	defaultGraphMicrosoftComURI = "https://graph.microsoft.com"
+)
+
+type MSGraphApplicationClient struct {
+	authorization.BaseClient
+}
+
+func newMSGraphApplicationClient(subscriptionId string) MSGraphApplicationClient {
+	return MSGraphApplicationClient{authorization.NewWithBaseURI(defaultGraphMicrosoftComURI, subscriptionId)}
+}
+
+func (client MSGraphApplicationClient) addPasswordPreparer(ctx context.Context, applicationObjectID string, displayName string, endDateTime date.Time) (*http.Request, error) {
+	pathParameters := map[string]interface{}{
+		"applicationObjectId": autorest.Encode("path", applicationObjectID),
+	}
+
+	parameters := struct {
+		PasswordCredential *passwordCredential `json:"passwordCredential"`
+	}{
+		PasswordCredential: &passwordCredential{
+			DisplayName: to.StringPtr(displayName),
+			EndDate:     &endDateTime,
+		},
+	}
+
+	preparer := autorest.CreatePreparer(
+		autorest.AsContentType("application/json; charset=utf-8"),
+		autorest.AsPost(),
+		autorest.WithBaseURL(client.BaseURI),
+		autorest.WithPathParameters("/v1.0/applications/{applicationObjectId}/addPassword", pathParameters),
+		autorest.WithJSON(parameters),
+		client.Authorizer.WithAuthorization())
+	return preparer.Prepare((&http.Request{}).WithContext(ctx))
+}
+
+func (client MSGraphApplicationClient) addPasswordSender(req *http.Request) (*http.Response, error) {
+	sd := autorest.GetSendDecorators(req.Context(), autorest.DoRetryForStatusCodes(client.RetryAttempts, client.RetryDuration, autorest.StatusCodesForRetry...))
+	return autorest.SendWithSender(client, req, sd...)
+}
+
+func (client MSGraphApplicationClient) addPasswordResponder(resp *http.Response) (result PasswordCredentialResult, err error) {
+	client.ResponseInspector = func(responder autorest.Responder) autorest.Responder {
+		return autorest.ResponderFunc(func(resp *http.Response) error {
+			err := responder.Respond(resp)
+
+			//fmt.Println(fmt.Sprintf("Request [ID: %v] returned HTTP status %d", resp.Header["Request-Id"], resp.StatusCode))
```

Comment:
Did you mean to leave this commented-out code?

Reply:
removed